### ISSUE TITLE
null Intl.defaultLocale

### ISSUE THREAD

**Issue by awhitford** [dart-lang/intl#280]
I have a Flutter app, and I seem to be running into a race-condition issue whereby code is referencing `Intl.defaultLocale`, but it is _`null`_ rather than `en_US` as expected.

Is there some operation that I need to _`await`_ on to ensure that the `Intl` package has completely initialized before I build a Screen?

**Comment by awhitford** [comment]
Interesting observation:  my problem went away when I replaced `Intl.defaultLocale` with `Intl.getCurrentLocale()`.  Shouldn't they be the same?  Why do they both exist?

**Comment by kinex** [comment]
I see this issue too in my app, `defaultLocale` is null (even though `Intl.systemLocale` has been set) causing a crash. This is a bug as the documentation for `Intl.defaultLocale` says "_The default locale. This defaults to being set from systemLocale_", but actual implementation is (intl-0.17.0\lib\src\global_state.dart):
```dart
String? get defaultLocale {
  var zoneLocale = Zone.current[#Intl.locale] as String?;
  return zoneLocale == null ? _defaultLocale : zoneLocale;
}
```
It should be something like
```dart
String get defaultLocale {
  var zoneLocale = Zone.current[#Intl.locale] as String?;
  return zoneLocale ?? _defaultLocale ?? systemLocale;
}
```

**Comment by alanknight-wk** [comment]
Yes, it's very confusing. I just spent a bunch of time tracking down a problem due to it, and I think the original author should be suitably punished. So there's a `defaultLocale`, which is initially null. There's a `systemLocale`, which is also initially null, because we don't know how to find the system locale in general. But if you import one of the platform-specific libraries and call `findSystemLocale` it sets it. And then if you call `getCurrentLocale()` it will set `defaultLocale = systemLocale`, but if you access `defaultLocale` directly it won't. But accessing `defaultLocale` does pay attention to the zone variable for locale, used by `withLocale()`.

I'm not sure of the ideal resolution, but I think it would probably be reasonable make Intl.defaultLocale always initialized to en_US, and maybe make findSystemLocale return a value rather than setting an internal variable. It's likely that anyone using findSystemLocale is going to want to pass that through some filtering before deciding what they really want the locale to be.
